lib/firebase/src/user.dart

```diff
@@ -99,13 +108,43 @@ sealed class ThcUser {
           _collection.doc(firestoreId).delete(),
       ]);
 
+  /// Update the profile picture URL for the user.
+  Future<void> updateProfilePicture() async {
+    if (type == UserType.director || type == UserType.admin) {
+      final picker = ImagePicker();
+      final pickedFile = await picker.pickImage(source: ImageSource.gallery);
+
+      if (pickedFile != null) {
+        final File image = File(pickedFile.path);
+        final String downloadUrl = await _uploadProfilePicture(id!, image);
+
+        // Update the user's profile picture URL in Firestore
+        await _collection.doc(firestoreId).update({'profilePictureUrl': downloadUrl});
+
+        // Update local profilePictureUrl
+        copyWith(profilePictureUrl: downloadUrl);
+      }
+    } else {
+      throw Exception('Only admins and directors can update profile pictures.');
+    }
+  }
+
+  Future<String> _uploadProfilePicture(String userId, File image) async {
+    final storageRef = FirebaseStorage.instance.ref();
+    final profilePicRef = storageRef.child('profile_pics/$userId.jpg');
+
+    await profilePicRef.putFile(image);
+
+    return await profilePicRef.getDownloadURL();
+  }
+
   /// {@macro ThcUser}
   ThcUser copyWith({
     UserType? type,
     String? id,
     String? name,
     String? email,
-    String? phone,
```

Comment:
Great catch here, by the way! Totally forgot to delete `phone` from `copyWith` when I removed it from the class 😅